### PR TITLE
Add ValidResourceType to [Resource]Container

### PR DESCRIPTION
**Pull request by ShivangiReja**
Add valid resource property that points at resource parent.

Example: https://github.com/Azure/azure-sdk-for-net/blob/feature/mgmt-track2/sdk/resourcemanager/Proto.Client/compute/AvailabilitySetContainer.cs#L25

Resolves: https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_workitems/edit/5414

### CODE REVIEW

File: src/AutoRest.CSharp/Mgmt/Generation/ResourceContainerWriter.cs [hunk 1]
```diff
@@ -21,6 +22,9 @@ public void WriteClient(CodeWriter writer, ResourceContainer resourceContainer)
                 using (writer.Scope($"{resourceContainer.Declaration.Accessibility} partial class {cs.Name}"))
                 {
                     WriteClientCtors(writer, resourceContainer);
+
+                    writer.WriteXmlDocumentationSummary($"Gets the valid resource type for this object");
+                    writer.Line($"protected {typeof(ResourceType)} ValidResourceType => {typeof(ResourceGroupOperations)}.{nameof(ResourceGroupOperations.ResourceType)};");
```
Comment (m-nash) [inline src/AutoRest.CSharp/Mgmt/Generation/ResourceContainerWriter.cs:27]: this should be pulled from the operationGroup.Parent()

**Review by m-nash — CHANGES_REQUESTED** [review]

File: samples/Azure.Management.Storage/Generated/BlobContainerContainer.cs [hunk 2]
```diff
@@ -14,5 +16,10 @@ public partial class BlobContainerContainer
         protected BlobContainerContainer()
         {
         }
+
+        private static readonly ResourceType ResourceType = "Microsoft.Storage/storageAccounts";
```
Comment (m-nash) [inline samples/Azure.Management.Storage/Generated/BlobContainerContainer.cs:20]: This should be removed for containers

File: samples/Azure.Management.Storage/Generated/BlobContainerContainer.cs [hunk 3]
```diff
@@ -14,5 +16,10 @@ public partial class BlobContainerContainer
...
+
+        /// <summary> Gets the valid resource type for this object. </summary>
+        protected ResourceType ValidResourceType => ResourceType;
```
Comment (m-nash) [inline samples/Azure.Management.Storage/Generated/BlobContainerContainer.cs:23]: this should be "=> [ParentResource]Operations.ResourceType;"

**Review by m-nash — CHANGES_REQUESTED** [review]

**Review by chamons — APPROVED** [review]

File: package.json [hunk 4]
```diff
@@ -3,7 +3,7 @@
   "private": true,
   "description": "package.json intended for in-repo use only, package.json used for publishing is located in src/AutoRest.CSharp/package.json",
   "devDependencies": {
-    "@microsoft.azure/autorest.testserver": "3.0.21",
-    "autorest": "3.1.5"
+    "@microsoft.azure/autorest.testserver": "3.0.19",
+    "autorest": "3.0.6339"
```
Comment (ShivangiReja) [inline package.json:7]: Ahh, I didn't do these changes. Something got messed up when I rebased from feature/v3.

File: samples/Azure.Management.Storage/Generated/OperationContainer.cs [hunk 5]
```diff
@@ -14,5 +16,8 @@ public partial class OperationContainer
         protected OperationContainer()
         {
         }
+
+        /// <summary> Gets the valid resource type for this object. </summary>
+        protected ResourceType ValidResourceType => "";
```
Comment (m-nash) [inline samples/Azure.Management.Storage/Generated/OperationContainer.cs:21]: why is this one an empty string?

File: samples/Azure.Management.Storage/Generated/StorageAccountContainer.cs [hunk 6]
```diff
@@ -14,5 +16,8 @@ public partial class StorageAccountContainer
         protected StorageAccountContainer()
         {
         }
+
+        /// <summary> Gets the valid resource type for this object. </summary>
+        protected ResourceType ValidResourceType => "any";
```
Comment (m-nash) [inline samples/Azure.Management.Storage/Generated/StorageAccountContainer.cs:21]: Create a bug to follow up on "any" as a valid resource type this shouldn't be here

**Review by m-nash — APPROVED** [review]

File: samples/Azure.Management.Storage/Generated/StorageAccountContainer.cs (hunk 6 above)
Comment (ShivangiReja) [reply samples/Azure.Management.Storage/Generated/StorageAccountContainer.cs:21]: Added an extra acceptance criteria [here](https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_workitems/edit/5405) to handle this.

File: samples/Azure.Management.Storage/Generated/OperationContainer.cs (hunk 5 above)
Comment (ShivangiReja) [reply samples/Azure.Management.Storage/Generated/OperationContainer.cs]: Fixed!

**Pull request merged**